regexec.c

```diff
@@ -3409,6 +3409,7 @@ S_find_byclass(pTHX_ regexp * prog, const regnode *c, char *s,
                             {
                                 DEBUG_TRIE_EXECUTE_r(
                                     Perl_re_printf( aTHX_ " - legal\n"));
+                                failed = 0;
```

Comment:
Not sure if in scope for this PR, but since `failed` is only used as a boolean, it could be declared as `bool failed` (with `failed = 1/0` changed to `failed = TRUE/FALSE`, respectively).